fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/rescheduleloan/service/LoanRescheduleRequestWritePlatformServiceImpl.java

```diff
@@ -638,6 +643,44 @@ public CommandProcessingResult reject(JsonCommand jsonCommand) {
         }
     }
 
+    @Override
+    public CommandProcessingResult undo(JsonCommand jsonCommand) {
+        try {
+            final var loanId = jsonCommand.entityId();
+            final var loan = loanRepositoryWrapper.findOneWithNotFoundDetection(loanId);
+            final var loanRescheduleRequests = loanRescheduleRequestRepository.findByLoanIdAndStatusEnum(loanId,
+                    LoanStatus.SUBMITTED_AND_PENDING_APPROVAL.getValue());
+
+            if (loanRescheduleRequests.isEmpty()) {
+                throw new LoanRescheduleRequestNotFoundException();
+            }
+
+            this.loanRescheduleRequestDataValidator.validateForUndoAction(jsonCommand);
+
+            this.platformSecurityContext.authenticatedUser();
+
+            loanRescheduleRequests.stream()
```

Comment:
I really hoped for you to recall the abstract schedule generator to regenerate the schedule afressh after undoing a reschedule.